src/Scaffolding/VS.Web.CG.Mvc/BlazorIdentity/BlazorIdentityGenerator.cs

```diff
@@ -0,0 +1,544 @@
+            }
+
+            var createUserClass = false;
+            // if an existing user class was determined from the DbContext, don't try to get it from here.
+            // Identity scaffolding must use the user class tied to the existing DbContext (when there is one).
+            if (string.IsNullOrEmpty(blazorIdentityModel.UserClassName))
+            {
+                if (string.IsNullOrEmpty(commandlineModel.UserClass))
+                {
+                    blazorIdentityModel.UserClassName = "IdentityUser";
+                    blazorIdentityModel.UserClassNamespace = "Microsoft.AspNetCore.Identity";
+                }
+                else
+                {
+                    var existingUser = await RoslynWorkspaceHelper.FindExistingType(
+                        Workspace,
+                        AssemblyLoadContextLoader,
+                        ProjectContext.AssemblyName,
+                        commandlineModel.UserClass);
+
+                    if (existingUser != null)
+                    {
+                        IdentityHelper.ValidateExistingUserType(existingUser);
+                        blazorIdentityModel.UserClassName = existingUser.Name;
+                        blazorIdentityModel.UserClassNamespace = existingUser.Namespace;
+                    }
+                    else
+                    {
+                        createUserClass = true;
+                        blazorIdentityModel.UserClassName = IdentityHelper.GetClassNameFromTypeName(commandlineModel.UserClass);
+                        blazorIdentityModel.UserClassNamespace = IdentityHelper.GetNamespaceFromTypeName(commandlineModel.UserClass)
+                            ?? defaultDbContextNamespace;
+                    }
+                }
+            }
+
+            var applicationUserModel = new IdentityApplicationUserModel
+            {
+                UserClassName = blazorIdentityModel.UserClassName,
+                UserClassNamespace = blazorIdentityModel.UserClassNamespace
+            };
+
+            if (createUserClass)
+            {
+                ExecuteApplicationUserTemplate(applicationUserModel);
+            }
+
+            if (createDbContext)
+            {
+                var dbContextModel = new IdentityDbContextModel
+                {
+                    DbContextName = blazorIdentityModel.DbContextName,
+                    DbContextNamespace = blazorIdentityModel.DbContextNamespace,
+                    UserClassModel = applicationUserModel
+                };
+
+                ExecuteDbContextTemplate(dbContextModel);
+                var connectionStringsWriter = new ConnectionStringsWriter(AppInfo, FileSystem);
+                connectionStringsWriter.AddConnectionString("DefaultConnection", dbContextModel.DbContextName, commandlineModel.DatabaseProvider);
+            }
+
+            //gather blazor identity files to generate
+            blazorIdentityModel.FilesToGenerate = new List<string>();
```

Comment:
Could you put this in an else, so you don't new it up if you're going to just replace it on line 268 or 288?

Reply:
addressed, check comment below